pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -744,6 +744,22 @@ func findNextInitContainerToRun(pod *v1.Pod, podStatus *kubecontainer.PodStatus)
 		return nil, nil, true
 	}
 
+	// If any of the main containers have status, then all init containers must
+	// after executed at some point in the past.  However, they could be removed
+	// from the container runtime now, and if we proceed, it would appear as if they
+	// never ran and will re-execute improperly.
+	for i := range pod.Spec.Containers {
```

Comment:
I might be very much off here.

This logic was added recently (#92614):
https://github.com/kubernetes/kubernetes/blob/6715318ee738668db0ddb84415f6926208761b9b/pkg/kubelet/kuberuntime/kuberuntime_manager.go#L531-L538

Looking at when `createPodSandbox` can be true, I wonder if one of the main containers may have a running state (https://github.com/kubernetes/kubernetes/blob/6715318ee738668db0ddb84415f6926208761b9b/pkg/kubelet/kuberuntime/kuberuntime_manager.go#L453) so this PR will break the logic as sandbox will not be created.